dask/dataframe/tests/test_groupby.py

```diff
@@ -2420,3 +2420,44 @@ def test_groupby_sort_true_split_out():
     with pytest.raises(NotImplementedError):
         # Cannot use sort=True with split_out>1 (for now)
         M.sum(ddf.groupby("x", sort=True), split_out=2)
+
+
+@pytest.mark.skipif(
+    not dd._compat.PANDAS_GT_100, reason="observed only supported for newer pandas"
+)
+@pytest.mark.parametrize("known_cats", [True, False])
+@pytest.mark.parametrize("ordered_cats", [True, False])
+@pytest.mark.parametrize("groupby", ["cat_1", ["cat_1", "cat_2"]])
+@pytest.mark.parametrize("observed", [True, False])
+def test_groupby_aggregate_categorical_observed(
+    known_cats, ordered_cats, agg_func, groupby, observed
+):
+    if agg_func in ["cov", "corr", "nunique"]:
+        pytest.skip("Not implemented for DataFrameGroupBy yet.")
+    if agg_func in ["sum", "count", "prod"] and groupby != "cat_1":
+        pytest.skip("Gives zeros rather than nans.")
```

Comment:
These all failed on master. Not sure if the test needs to be improved. Or if these are expected differences.

Reply:
@TomAugspurger ? 
I can certainly see why zero is convenient...

Reply:
Who gives zeros and who gives NaNs?

The expected count is always not-NA. The expected sum of an empty array (unobserved category) should be 0 and the expected product should be 1.

Reply:
Sorry that was very unclear. pandas prod gives nans, dask prod gives 1s. pandas sum gives nans, dask sum gives 0s, pandas count gives nans, dask count gives 0s. This is when I run this test without the observed kwarg on master. 

for example count:

```python
(Pdb) expected
             value_1
cat_1 cat_2         
A     1          1.0
      2          NaN
      3          NaN
B     1          NaN
      2          1.0
      3          NaN
C     1          NaN
      2          NaN
      3          NaN
D     1          NaN
      2          NaN
      3          NaN
E     1          NaN
      2          NaN
      3          NaN
(Pdb) actual.compute()
             value_1
cat_1 cat_2         
A     1          1.0
      2          0.0
      3          0.0
B     1          0.0
      2          1.0
      3          0.0
C     1          0.0
      2          0.0
      3          0.0
D     1          0.0
      2          0.0
      3          0.0
E     1          0.0
      2          0.0
      3          0.0

```

Reply:
Since these skips are just capturing what is already on master, I feel like it's a separate issue to try to fix the discrepancies.